src/dnmd/streams.c

```diff
@@ -5,6 +5,13 @@ bool try_get_string(mdcxt_t* cxt, size_t offset, char const** str)
     assert(cxt != NULL && str != NULL);
 
     mdstream_t* h = &cxt->strings_heap;
+
+    if (h->size == 0 && offset == 0)
```

Comment:
Yep. The manual tests where we build a minimal image effectively require this, otherwise they can end up failing to read valid empty-string columns.

Reply:
Yep. Each of the heaps has a well-defined 0th element, and providing this support makes it possible to add a row with a 0-index into a heap (like an empty name) and have it still be a readable column. Without this, reading the name column would fail with "corrupt image".